sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/NestedDataOperatorConversions.java

```diff
@@ -276,7 +276,8 @@ public SqlRexConvertlet createConvertlet(PlannerContext plannerContext)
               call.operand(0),
               call.operand(1)
           );
-        } else if (SqlTypeName.APPROX_TYPES.contains(sqlType.getSqlTypeName())) {
+        } else if (SqlTypeName.APPROX_TYPES.contains(sqlType.getSqlTypeName()) ||
+                   SqlTypeName.DECIMAL.equals(sqlType.getSqlTypeName())) {
```

Comment:
perhaps invert these?  Equality is faster than `.contains()` so will be better to check the equality first and then do the `.contains()` instead of forcing the `.contains()` to fail first before doing the equality.